osh/cmd_eval.py

```diff
@@ -304,6 +304,9 @@ def _PrefixBindingsPersist(cmd_val):
             return True
         elif case(cmd_value_e.Argv):
             cmd_val = cast(cmd_value.Argv, UP_cmd_val)
+            if len(cmd_val.argv) == 0:
+                return True
```

Comment:
Hm it's somewhat surprising to me that the binding persists - I was thinking it should be `return False` !!!

```
$ bash -c 'FOO=bar $undef; echo FOO=$FOO'
FOO=bar


$ bash -c 'FOO=bar echo; echo FOO=$FOO'  # does NOT persist
FOO=
```

I would have thought FOO=bar goes out of scope in the first case, like the second

It looks like the spec test covers it, but how about adding a comment about this behavior?

Reply:
Done!